src/resource/worker-control.c

```diff
@@ -797,8 +797,7 @@ int worker_control_get_new_worker(sid_resource_t *worker_control_res, struct wor
 			                    worker_control->channel_spec_count) < 0)
 				goto out;
 
-			if (worker_control->init_cb_spec.cb)
```

Comment:
Actually, just recalled, this provided a way for doing any preparation/cleanup right before the `exec` is called. There shouldn't be `NULL` dereference - the worker init fn is a property of `worker_control` which is inherited in the child process and still available before we call `exec`.

There's just one thing to care about - if we're preparing an external worker, the `init_cb_spec.cb` (the `_worker_init_fn` in our concrete case) is called with NULL for the worker resource: `_worker_init_fn(NULL, ...)`. So the `worker` resource won't be available there, but still we have a chance to do some preparation/cleanup before we do `exec`, if needed, even without the worker resource.

Reply:
Oops. Yeah, I forgot that the external worker's `init_cb_spec.cb` callback wouldn't be `_worker_init_fn()`. I'll change the patch. Should I leave in the code to destroy all the resources starting from the top, under the assumption that we'll always want to do that, and the callback can copy-out any resources that it wants to keep. Or should we leave the destroying for the callback to do, like it does in `_worker_init_fn()`.

This does make a nice segue to something that has been bothering me. I assume the way that external workers will end up working is that SID will have multiple worker_control resources, each dedicated to a different kind of worker. All the plumbing to make SID pick the correct kind of worker is still work to be done then. This means that we can figure out how to have the sid main() function determine which worker spawned a new process later as well. It could be something as simple as a new `worker_control` bool called `active` which would be set after the `fork()`.